File: src/main/java/com/digsol/turnomed/mapper/UsersMapper.java
package com.digsol.turnomed.mapper;

import com.digsol.turnomed.model.Users;
import org.apache.ibatis.annotations.Mapper;
import org.springframework.stereotype.Component;

import java.util.List;

@Mapper
@Component
public interface UsersMapper {
    int deleteByPrimaryKey(Integer id);

    int insert(Users record);

    int insertSelective(Users record);

    Users selectByPrimaryKey(Integer id);

    int updateByPrimaryKeySelective(Users record);

    int updateByPrimaryKey(Users record);

    List<Users> listAllUsers();

    Users getUserByEmail(String email);

}

File: src/main/kotlin/com/digsol/turnomed/mapper/UsersMapper.kt
package com.digsol.turnomed.mapper

import com.digsol.turnomed.model.Users
import org.apache.ibatis.annotations.Mapper
import org.springframework.stereotype.Component

@Mapper
@Component
interface UsersMapper {
    fun deleteByPrimaryKey(id: Integer): Int

    fun insert(record: Users): Int

    fun insertSelective(record: Users): Int

    fun selectByPrimaryKey(id: Integer): Users

    fun updateByPrimaryKeySelective(record: Users): Int

    fun updateByPrimaryKey(record: Users): Int

    fun listAllUsers(): List<Users>

    fun getUserByEmail(email: String): Users
}
